registry/storage/driver/gcs/gcs.go

```diff
@@ -640,33 +652,29 @@ func (d *driver) Stat(context context.Context, path string) (storagedriver.FileI
 
 // List returns a list of the objects that are direct descendants of the
 // given path.
-func (d *driver) List(context context.Context, path string) ([]string, error) {
+func (d *driver) List(ctx context.Context, path string) ([]string, error) {
 	var query *storage.Query
 	query = &storage.Query{}
 	query.Delimiter = "/"
 	query.Prefix = d.pathToDirKey(path)
 	list := make([]string, 0, 64)
```

Comment:
Not new, but (also see my other comment); do we know where the initial `64` capacity comes from? Is that just a randomly picked value, and should this be based on actual (expected number of) results instead?

Reply:
I vaguely remember this was a bit of a mystery to me in past and now you've revived the memories. It's hard to think of an "expected number of results other than an educated guess here - `List` lists the storage keys which can be arbitrarily long (or short). `64` is a slice `capacity` parameter not the `length` so I see this as a pre-allocated buffer guesstimate that might help us reap some benefits should the number of listed storage keys is `>64` (no need to reallocate the slice until we exceed the count of `64`); we could discuss performance implications of this for sure, but I don't see this as a major problem at the moment.